### ISSUE TITLE
Support model validation during training for the PyTorch wrapper

### ISSUE THREAD

**Issue by fcogidi** [VectorInstitute/cyclops#365]
- Split the dataset into training and validation sets in the [`_train_loop`](https://github.com/VectorInstitute/cyclops/blob/main/cyclops/models/wrappers/pt_model.py#L646) method or accept a predefined split when the `fit` method is called.
- Report the validation loss.
- Use the validation loss to track the best model.
